### PR TITLE
cleanup and comment Calendar code. Fixed adding same user bug. Setup Meeting listener and removed setState methods on Event updates

### PR DESCRIPTION
**Pull request by Full-lifey**
# Description

Describe change or new feature here.

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

File: src/components/calendar/MeetingModal.js [hunk 1]
```diff
@@ -35,33 +35,33 @@ const MeetingModal = props => {
...
     let newParticipants = {
       participantUIDs: meeting.participantUIDs || [],
       participantNames: meeting.participantNames || []
     };
+    //* Checks if new participants have been added and if they are new, adds them to newParticipants obj
     if (participants.length) {
       participants.map(participant => {
-        newParticipants.participantUIDs.push(participant.uid);
-        newParticipants.participantNames.push(participant.fullName);
+        if (!newParticipants.participantUIDs.includes(participant.uid)) {
```
Comment (Full-lifey) [inline src/components/calendar/MeetingModal.js]: Added to fix a bug where the same user can be added multiple times.

File: src/components/shared/components/SearchResults.js [hunk 2]
```diff
@@ -47,10 +48,11 @@ export default function SearchResults(props) {
 
   useEffect(() => {
     setSearchResults(props.searchResults);
-  });
+  }, [props.searchResults]);
```
Comment (Full-lifey) [inline src/components/shared/components/SearchResults.js]: Added so useEffect doesn't run every render

File: src/components/shared/components/SearchResults.js [hunk 3]
```diff
@@ -47,10 +48,11 @@ export default function SearchResults(props) {
...
 
+  //* Pulls previously added participants(if any) and adds selected participants to MeetingModal State and adds to display
   const saveParticipants = async () => {
-    let invitedUsers = [];
+    let invitedUsers = props.participants;
```
Comment (Full-lifey) [inline src/components/shared/components/SearchResults.js]: Now it pulls in previously selected users if more than one is searched.

File: src/components/calendar/Calendar.js [hunk 4]
```diff
@@ -64,100 +63,63 @@ class Calendar extends React.Component {
...
   };
 
+  // Edits current meeting
   editMeeting = async meeting => {
     console.log("meeting arg into editMeeting", meeting);
     try {
       //* updating meeting in firebase
       const meetingRef = firestore.collection("meetings").doc(meeting.id);
       await meetingRef.update(meeting);
-      //* edit event in events array
```
Comment (micahjones13) [inline src/components/calendar/Calendar.js]: Jeez, you weren't kidding when you said you slimmed it down. Nice work

**Pull request merged**